add workshop for git day 1 (#80)
CodeYourFuture/CYF-Workshops: git-day-1/README.md

## Part 1 -  📸 Git week 1 workshop

> In this workshop, you'll learn the following: 

Author: I've put these learning objectives into front matter - but they're not being rendered

> In this workshop, you'll learn the following: 

Reviewer: ```objectives
- [ ] like this
```

> In this workshop, you'll learn the following: 

Reviewer: er. ```objectives

1. Get a local copy of a repository on your local machine.
2. Open this local repository using VSCode.
3. Use `pwd` in your terminal to check you're in the right place.

 
## Recap - 🌳 creating a branch

In this section, you'll need to create a branch so that your local git repository looks like this:

```mermaid
---
title: cyf-demo-repo
---
gitGraph
   commit
   commit
   commit
   branch  "week-1-workshop"
```

## New skill 🧰 - Creating a commit 📸

We have the following goal in this section:

> 🎯 Goal: Make some **local** changes and create a **local commit** on our branch.

You'll need to carefully follow through the instructions in this task.

### Do this section in pairs 
### ⏲️ ( 25 mins )

Our Git timeline starts off with some commits like this:
```mermaid
---
title: cyf-demo-repo
---
gitGraph
   commit
   commit
   commit
   branch  "week-1-workshop"
```

We're going to figure out how to 

- make local changes and commit them to our branch "week-1-workshop",

so our history ends up looking something like this:

```mermaid
---
title: cyf-demo-repo
---
gitGraph
   commit
   commit
   commit
   branch "week-1-workshop"
   commit
   commit
```

> where the most recent commits contain our recent work.

Reviewer: ```suggestion
The most recent commits contain our most recent work.
```

> To create a commit, you're going to go through the following steps:

Reviewer: ```suggestion
Create a commit using these steps:
```

1. Make a change to a file
2. View the local changes
2. Stage the changes
3. Creating the commit

### 🗄️ 1. Make a change to a file

> 1. Open up your local github repo `cyf-demo-repo`.

Reviewer: ```suggestion
1. Open up your local repo `cyf-demo-repo`.
```

It's a bit confusing to call it a local github repo

2. Go to the Explorer section of VSCode

> 3. Locate `file.txt` and edit the file with the answer to the questions.

Reviewer: ```suggestion
3. Find `file.txt` and edit the file with the answer to the questions.
```

We want to view the changes we've just made to our working directory.

1. Go to the **Source Control** tab in VSCode.

![vscode-sidebar-source-control](vscode-sidebar-source-control.png)

2. Go to the Changes section and look at the changes for the file.
3. Try editing the file again in the Explorer tab and check to see the update is visible in the Source Control panel

### 🟢 3. Stage the changes

We need to tell Git which changes we want to be part of our next commit. 

We do this by **staging our changes**.

1. Go to the file `README.md` and click on the **+**.

👓 Notice what happens when you carry out step 1.

Explain what you think will happen if you don't stage anything in your working directory.

2. View the **Staged Changes** area in your **Source Control** panel.

### 📸 4. Create the commit

Once we've staged changes, then we can **commit** these changes.

📝 Check the number of commits you have locally. They should still be the same as before.

1. Enter a commit message describing briefly what you did in your commit.
2. Click Commit to create the Git commit.

📝  Now re-check the number of commits you have locally.

### When you're finished...

> > ❗ Once you've completed this **commit**, you can swap roles in your pair.

Reviewer: ```suggestion
> ❗ Once you've completed this **commit**, swap roles in your pair.
```

> Choose another file and then go through the steps in [Creating a commit section](#creating-a-commit)

--- 

### Take a break... 🍫

---

## Part 2

### 🫸 Pushing (15 mins)

> After committing your work locally, you'll have a local branch that looks like this

Reviewer: ```suggestion
After committing your work on your machine, you'll have a local branch that looks like this
```

```mermaid
---
title: cyf-demo-repo
---
gitGraph
   commit
   commit
   branch "week-1-workshop"
   commit
   commit
```

> 📋 Double-check you've been committing to your branch `week-1-workshop` branch and not your `main` branch. 

Reviewer: BTW I just updated the front end so you can use hugo note markup in comments and it will show up on the curriculum

However, our remote fork only has a main branch:

```mermaid
---
title: cyf-demo-repo
---
gitGraph
   commit
   commit
```

> In other words, we've not added our local branch on to the remote fork on Github.

Reviewer: ```suggestion
In other words, we have _not_ added our local branch on to the remote fork on Github.
```

Avoid strong regionalisms like we've not

## Creating a fork 🍴

### 🕹️ Recap activity 

- Trainees do this section in pairs
- ⏲️ 5 mins

Use the guidelines from the prep section of this week to create a fork.

1. On one person's computer, fork this repo: https://github.com/CodeYourFuture/cyf-demo-repo
2. 📝 Double-check the URL of your forked repo. How can you tell the fork was successfully created?

### Whole group discussion/demo 🧵 🗣️

- Whole class discussion
- ⏲️ 5 mins
- A volunteer will need to facilitate this section

A volunteer will need to navigate and let the group guide them in creating a fork as per the steps in the previous section.

## 🌀 Cloning a repository 

### 🕹️ Recap activity 

- Trainees do this section in pairs
- ⏲️ 5 mins

#### 📋 Check your understanding

Before continuing, try answering the following:

❓ what is the difference between a **fork** and a **clone**.

Remember to check your answer before continuing.

1. Get a local copy of a repository on your local machine.
2. Open this local repository using VSCode.
3. Use `pwd` in your terminal to check you're in the right place.

> Now clone your forked repository from Github onto your local machine.

Reviewer 2: Sorry for being late to the party but I'm struggling to make sense of this exercise. I'm happy to submit a PR with fixes but I wanted to understand the intention first.

> Get a local copy of a repository on your local machine.

"Get" is a bit ambiguous here: does this mean to open an existing repo? Or to clone a repo?

If it's the latter, which repo should they clone? Any repo? I think I would prefer a specific repo, as it'll just slow down the exercise to pick a repo to clone.

> Now clone your forked repository from Github onto your local machine.

Assuming the points above are to clone a forked repo, this is just repeating the same point?

Or is this line just in the wrong place - should it be above the list?
